File: 3170. Lexicographically Minimum String After Removing Stars/clearStars.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
    public:
        string clearStars(string s) {
            string str;
            int len = s.size();
            vector<vector<int>>idxs(26);
            vector<bool>removed(len,false);
    
            for(int i=0 ; i<len; i++){
                if(s[i] != '*'){
                    idxs[s[i]-97].push_back(i);
                }
                else{
                    for(int chr=0 ; chr<26 ; chr++){
                        if(idxs[chr].size() > 0){
                            int poped = idxs[chr].back();
                            idxs[chr].pop_back();
                            removed[poped] = true;
                            break;
                        }
                    }
                    removed[i] = true;
                }
            }
    
            for(int i=0 ; i<len; i++){
                if(!removed[i]){
                    str+= s[i];
                }
            }
            return str;
        }
    };

File: 3170. Lexicographically Minimum String After Removing Stars/clearStars.js
/**
 * @param {string} s
 * @return {string}
 */

var clearStars = function (s) {
    let len = s.length
    let idxs = new Array(26).fill().map(() => [])
    let arr = s.split("")
    //a => 97

    for (let i = 0; i < len; i++) {
        let code = arr[i].charCodeAt(0) - 97
        if (arr[i] !== '*') {
            idxs[code].push(i)
        }
        else {
            for (let chr = 0; chr < 26; chr++) {
                if (idxs[chr].length > 0) {
                    let poped = idxs[chr].pop()
                    arr[poped] = ''
                    break;
                }
            }
            arr[i] = ''
        }
    }
    let str = arr.join('')

    return str

};